tracing-subscriber/src/reload.rs

```diff
@@ -36,7 +36,34 @@ pub struct Layer<L, S> {
     _s: PhantomData<fn(S)>,
 }
 
-/// Allows reloading the state of an associated `Layer`.
+/// Allows reloading the state of an associated [`Layer`](crate::layer::Layer).
+///
+/// # Examples
+///
+/// Reloading a [`Filtered`](crate::filter::Filtered) layer to change the filter at runtime.
+///
+/// ```
+/// # use tracing::info;
+/// # use tracing_subscriber::{filter,fmt,reload,Registry,prelude::*};
+/// # fn main() {
+/// let (filtered_layer, reload_handle) =
+///     reload::Layer::new(fmt::Layer::default().with_filter(filter::LevelFilter::WARN));
+/// #
+/// # // specifying the Registry type is required
+/// # let _: &reload::Handle<filter::Filtered<fmt::Layer<Registry>,
+/// # filter::LevelFilter, Registry>,Registry>
+/// # = &reload_handle;
+/// #
+/// info!("This will be ignored");
+/// reload_handle.modify(|layer| *layer.filter_mut() = filter::LevelFilter::INFO);
+/// info!("This will be logged");
+/// # }
+/// ```
+///
+/// **Warning:** The [`Filtered`](crate::filter::Filtered) type currently can't be changed
+/// at runtime via the [`Handle::reload`] method.
+/// Use the [`Handle::modify`] method to change the filter instead.
+/// (see <https://github.com/tokio-rs/tracing/issues/1629>)
```

Comment:
i would probably put this note in the documentation for the `Handle::reload` method, so that it's obvious to people who are looking at the docs for that method where they can't use it.